### ISSUE TITLE
User-defined multi-categorization of tracks for filtering

### ISSUE THREAD

**Issue by lazka** [quodlibet/quodlibet#1221]
Original [issue 1221](https://code.google.com/p/quodlibet/issues/detail?id=1221) created by destinyjwoods on 2013-07-22T03:52:25.000Z:

This is sort of two features in one, but they are inextricably linked; the first is made most useful by the second and the second is made most useful by the first, if not depending on it.  Hence I've described the whole system below.  Let me know if I absolutely _must_ sort them out into two feature requests.

I would like to see the ability (probably via internal tags--but ones the user can set) to &quot;categorize&quot; music in user-defined categories, and then filter based on that.  For instance, _independent of genre_, I would like to mark some of my music as instrumental vs. vocal, or as &quot;meditative&quot; or &quot;relaxing&quot;.  I, personally, use the word &quot;contemporary&quot; to categorize anything with a beat (which may or may not overlap with instrumental), and a song categorized folk could also be categorized contemporary AND vocal or instrumental in addition.  This is clearly not what most people use--which is why a user-defined method will work best, to accommodate people's idiosyncratic ways of thinking and organizing music.  What I'm referring to is a system of categorizing and working with music that is far more complex than the single-genre mindset.

I don't think that changing the genre system is a good idea; some players have difficulty with multiple genres, the system I'm referring to is likely to be very user-specific (my definition of &quot;meditative&quot; is not necessarily anyone else's, and I use terms that no one else might use; I'm sure I'm not the only one), and in writing something to the actual file tags, it eliminates the possibility of importing music that cannot or should not be changed (such as a song that's being seeded in a torrent; changing the tags would cause errors with the hash).  The concept of internal tags as QL has set up could be extended to cover this.

Inputting the tags could be done by checkbox (I'd prefer that, as it's far easier to quickly check, check, check tags that I use over and over again rather than type and tab-complete or something else complicated), but maybe there's a better way.  The user would have control over which tags exist--adding and deleting them as need be.

Why would this system be useful and desirable to many people?  Say that someone wants to filter their music by categories.  Currently I have no way to easily filter my music to play only religious Christmas music (if I tag all Christmas music specifically religious vs. secular, then there's no way to sort all Christmas together, or vice versa), or to filter out folk music with singing from folk music that does not have singing (which I might not want to do _all_ the time--other times simply playing anything folk); in college I frequently wanted to sort out instrumental music _without_ a beat for background study music, but sometimes include all instrumental.  (That's just the way _I_ like to sort my music; I'm sure others would have different combinations of categories they would like to sort!)

With a large number of tracks (and QL boasts to be solid for large numbers of tracks, which I have), manually sorting to playlists every possible combination would be a nightmare (most users would just give up).  With an initial investment in categorizing the music the way the user thinks of it, the user could then filter just the appropriate set of categories for their particular situation.  This would give the user unprecedented control of their music.  (If the filtering is set up to allow Boolean AND as well as OR, it would beat out even the old dBpowerAMP Audio Player, which allowed multi-genre tagging in-database and filtering by OR.)  Filtering could also be extended to combine categories with information such as year, artist, even bpm, making it very useful for DJs.  (Filtering should be yet another way to browse the library--this one allowing users to enter in information in various boxes by embedded and/or internal tags.)

In short, this would give users of QL more control over their music, control that few players have even come close to giving.  This control is useful not just for control's sake, but will save users massive amounts of time browsing through trying to manually create the right playlists, and allow them to quickly sort their music for specific situations; it would allow the player to fit _them_ instead of requiring them to adapt to the player.  It might be a bit of work; I think the outcome will be well-worth it.
